### PR TITLE
[envsec] Add (hidden) envsec init command

### PR DESCRIPTION
**Pull request by mikeland73**
## Summary

TSIA

## How was it tested?

`devbox envsec init`

### CODE REVIEW

File: internal/boxcli/envsec.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	box.Config().SetStringField("EnvFrom", "envsec")
+	if err := box.Config().SaveTo(box.ProjectDir()); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return envsec.EnsureInitialized(cmd.Context(), box.ProjectDir())
```
Comment (savil) [inline internal/boxcli/envsec.go:57]: can we swap the order of envsec.EnsureInitialized and modifying the devbox.json?

i.e. I feel we should only modify the devbox.json once we know envsec has been properly initialized.

File: internal/devconfig/env.go [hunk 2]
```diff
@@ -18,7 +18,8 @@ func (c *Config) ComputedEnv(
 	if c.IsEnvsecEnabled() {
 		env, err = envsec.Env(ctx, projectDir)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error reading secrets from envsec: %s\n\n", err)
+			ux.Ferror(os.Stderr, "Error reading secrets from envsec: %s\n\n", err)
```
Comment (savil) [inline internal/devconfig/env.go:21]: Given that we are continuing on, should this be a `ux.Fwarning` instead?

Comment (savil) [reply internal/devconfig/env.go:21]: i.e. something like:

```
ux.Fwarning(os.Stderr, "Computing Devbox environment without envsec secrets. There was an error from envsec: %s\n\n", err)
```

**Review by savil — APPROVED** [review]

File: internal/devconfig/env.go (hunk 2 above)
Comment (mikeland73) [reply internal/devconfig/env.go:21]: I'll can switch to warning. My motivation is that it's a fairly big error (not really a warning) so I wanted to make sure it's clear. But generally we don't use errors for commands that end in success.

**Pull request merged**